pkg/vmprovider/providers/vsphere/contentlibrary/content_library_cache.go

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2023 VMware, Inc. All Rights Reserved.
```

Comment:
Big comment incoming on this.

Reply:
See this comment -- https://github.com/vmware-tanzu/vm-operator/pull/47#pullrequestreview-1277855425

Reply:
I actually thought about making this cache generic or using a 3rd party cache library, however, I made it specific to the content-library as I didn't see any obvious usecase for storing other objects atm. I am taking a deeper look at your implementation above and will try adopting it in this PR. Thanks again for sharing it.

Reply:
Updated this PR with your generic type cache included.